### ISSUE TITLE
what should notice/ignore do when given a value outside the valid range?

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#1128]
At the moment notice and ignore filters that are outside the valid range get clamped to the minimum or maximum value. In the example below I show an example where we notice the 2000 to 3000 keV range which - for the Chandra data - doesn't match anything but ends up selecting the last bin.

I can understand why this happens (I believe we internally clamp the channel range to the valid range so that things like `notice(2, 200)` where 200 is outside the supported range doesn't cause a problem), but I believe it is problematic in general (such as the examples below, and I've noticed this oddity in other work I've done recently). I'm not sure what we sould do about this, hence this reminder.

```
In [1]: from sherpa.astro.io import read_pha

In [2]: pha = read_pha('sherpa-test-data/sherpatest/3c273.pi')
WARNING: systematic errors were not found in file 'sherpa-test-data/sherpatest/3c273.pi'
statistical errors were found in file 'sherpa-test-data/sherpatest/3c273.pi'
but not used; to use them, re-read with use_errors=True
read ARF file sherpa-test-data/sherpatest/3c273.arf
read RMF file sherpa-test-data/sherpatest/3c273.rmf
WARNING: systematic errors were not found in file 'sherpa-test-data/sherpatest/3c273_bg.pi'
statistical errors were found in file 'sherpa-test-data/sherpatest/3c273_bg.pi'
but not used; to use them, re-read with use_errors=True
read background file sherpa-test-data/sherpatest/3c273_bg.pi

In [3]: pha.units
Out[3]: 'energy'

In [4]: pha.mask
Out[4]: True

In [5]: pha.notice(2000, 3000)

In [6]: pha.mask
Out[6]:
array([False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
        True])

```

Note that the last bin has been set `True`. I repeat this below but with a "valid" noticed range, and we can see that the last bin again gets selected:

```
In [9]: pha.notice(None, None)

In [10]: pha.notice(0.5, 6)

In [11]: pha.mask
Out[11]:
array([False, False, False,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True, False,
       False])

In [12]: pha.notice(2000, 3000)

In [13]: pha.mask
Out[13]:
array([False, False, False,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True, False,
        True])
```

**Comment by DougBurke** [comment]
So, if we have a PHA file with an associated energy range of 0.1 to 15 keV, and we're filtering in energy units, then we'd want

| Filter | Equivalent to |
| ----- | --- |
| pha.notice(0.5, 20) | 0.5-15 |
| pha.notice(0.01, 0.5) | 0.1-0.5 |
| pha.notice(0.01, 20) | 0.1-15 |

and this is what we have at the moment. The issue is what to do with the following, where I claim the "equivalent to" behavior is unexpected:

| Filter | Equivalent to |
| ----- | --- |
| pha.notice(0.01, 0.1) | 0.5-0.5 |
| pha.notice(20, 20) | 15-14 |

It may be possible to add an explicit check for `notice` and `ignore` to say if both the `lo` and `hi` values are smaller than (or higher than) the limit value then we do nothing.

**Comment by DougBurke** [comment]
This has been addressed by #1127 and now the code output is

```python
In [1]: from sherpa.astro.ui import *
WARNING: imaging routines will not be available,
failed to import sherpa.image.ds9_backend due to
'RuntimeErr: DS9Win unusable: Could not find xpaget on your PATH'

In [2]: load_pha('sherpa-test-data/sherpatest/3c273.pi')
WARNING: systematic errors were not found in file 'sherpa-test-data/sherpatest/3c273.pi'
statistical errors were found in file 'sherpa-test-data/sherpatest/3c273.pi'
but not used; to use them, re-read with use_errors=True
read ARF file sherpa-test-data/sherpatest/3c273.arf
read RMF file sherpa-test-data/sherpatest/3c273.rmf
WARNING: systematic errors were not found in file 'sherpa-test-data/sherpatest/3c273_bg.pi'
statistical errors were found in file 'sherpa-test-data/sherpatest/3c273_bg.pi'
but not used; to use them, re-read with use_errors=True
read background file sherpa-test-data/sherpatest/3c273_bg.pi

In [3]: pha = get_data()

In [4]: pha.units
Out[4]: 'energy'

In [5]: pha.mask
Out[5]: True

In [6]: pha.notice(2000, 3000)

In [7]: pha.mask
Out[7]: True
```

Notice that the mask is still `True`.

```python
In [8]: pha.notice(None, None)

In [9]: pha.mask
Out[9]: True

In [10]: pha.notice(0.5, 6)

In [11]: pha.mask
Out[11]:
array([False, False, False,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True, False,
       False])

In [12]: pha.notice(2000, 3000)

In [13]: pha.mask
Out[13]:
array([False, False, False,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True, False,
       False])
```

Notice that the mask hasn't changed with the "invalid" notice call.
